src/nspawn/nspawn.c

```diff
@@ -2725,6 +2750,35 @@ static int inner_child(
         return log_error_errno(r, "execv() failed: %m");
 }
 
+static int setup_sd_notify_child(void) {
+        static const int one = 1;
+        int fd = -1;
+        union sockaddr_union sa = {
+                .sa.sa_family = AF_UNIX,
+        };
+        int r;
+
+        fd = socket(AF_UNIX, SOCK_DGRAM|SOCK_CLOEXEC|SOCK_NONBLOCK, 0);
+        if (fd < 0) {
+                safe_close(fd);
```

Comment:
well, you just checked that fd is negative, i.e. invalid, there's no point in closing it here, you can drop that line.